src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -511,6 +511,8 @@ memory::ptr primitive_inst::allocate_output(engine& _engine, memory_pool& pool,
 
     std::function<bool(const program_node&)> user_requesting_mem_reuse_false = [&user_requesting_mem_reuse_false](const program_node& node) {
         for (auto& user : node.get_users()) {
+            if (user->is_dynamic())
+                continue;
```

Comment:
I think even if one of the user is dynamic it should return true.

Reply:
Fixed